include/zcbor_common.h

```diff
@@ -14,12 +14,21 @@
 /** Convenience type that allows pointing to strings directly inside the payload
  *  without the need to copy out.
  */
-struct zcbor_string
-{
+struct zcbor_string {
 	const uint8_t *value;
 	size_t len;
 };
 
+
+struct zcbor_string_fragment {
```

Comment:
Do we really need this struct definition exposed in the header file? If possible, would it not be better to have a blank declaration here, and then have the definition of the type in an internal header file?

This to avoid users operating directly on members of the struct so that their code won't be future proof.

If you have multiple types defined here which are typically accessed through the proper API then I guess there is little point in moving just this definition to an internal header file.

Reply:
I think users might want to access the member variables. I will add a bit of docs telling users not to modify anything.